src/main/java/se/liu/ida/hefquin/engine/wrappers/lpgwrapper/utils/VariableReplacementVisitor.java

```diff
@@ -168,6 +167,21 @@ public void visitNodeMatch(final NodeMatchClause ex) {
         stack.push(new NodeMatchClause((CypherVar) stack.pop()));
     }
 
+    @Override
+    public void visitPathMatch(PathMatchClause ex) {
+        final Deque<PathMatchClause.EdgePattern> edges = new ArrayDeque<>();
+        for (int i = ex.getEdges().size(); i > 0; i--) {
+            assert stack.peek() instanceof CypherVar;
+            final CypherVar right = (CypherVar) stack.pop();
+            assert stack.peek() instanceof CypherVar;
+            final CypherVar edge = (CypherVar) stack.pop();
+            assert stack.peek() instanceof CypherVar;
+            final CypherVar left = (CypherVar) stack.pop();
+            edges.push(new PathMatchClause.EdgePattern(left, edge, right, ex.getEdges().get(i).getDirection()));
```

Comment:
After making the member variables of `PathMatchClause.EdgePattern` all public, you can change this line to:
```suggestion
            edges.push( new PathMatchClause.EdgePattern(left, edge, right, ex.getEdges().get(i).direction) );
```